### DISCUSSION TITLE
React-query cache doesn't persist on page refresh

### DISCUSSION THREAD

**Discussion by DoneDeal0** [TanStack/query#2138] [Q&A]
I would like to set a 24 hours cache once a useQuery request has succeeded.

But as soon as I refresh the page, the cache is gone. I see it because I console.log a message each time the route is hit on my server.

How to prevent this behaviour and implement a real cache?

Here is the code:

```js
import { useQuery } from "react-query";
import { api } from "./config";

const _getUser = async () => {
  try {
    const res = api.get("/get-user");
    return res;
  } catch (err) {
    return err;
  }
};

export const getUser = () => {
  const { data } = useQuery("contact", () => _getUser(), {
    cacheTime: 1000 * 60 * 60 * 24,
  });
  return { user: data && data.data };
};


// then in the component:
  const { user } = getUser();

return (
    <div >
      hello {user?.name}
    </div>
```

I've also tried to replace cacheTime by staleTime.

**Comment by TkDodo** [comment]
if you reload the browser, the cache is gone because the cache lives in-memory. If you want a persistent cache, you can try out the (experimental) persistQueryClient plugin: https://react-query.tanstack.com/plugins/persistQueryClient

**Reply by DoneDeal0** [reply]
Thank you! (I also upvoted you on stackoverflow ;)).

**Comment by g-coutos** [comment]
@TkDodo if I may ask, why this plugin is not avaiable anymore on v5?

**Reply by TkDodo** [reply]
We haven't removed it 🤷‍♂️
